presto-main/src/main/java/io/prestosql/execution/QueryStateMachine.java

```diff
@@ -787,7 +787,13 @@ public void onFailure(Throwable throwable)
 
     private void transitionToFinished()
     {
-        cleanupQueryQuietly();
+        try {
+            metadata.cleanupQuery(session);
```

Comment:
I don't think we can make this change.  If this is a query in an auto-commit transaction, then the query is already committed and failure is no longer allowed.  Is there a reason, we need to be able to fail here?

Reply:
This is related to the comment i made here: https://github.com/prestosql/presto/pull/1034/files#r341981670

We need to fail the query if transaction commit fails.

Reply:
One option is that we call metadata.cleanupQuery before autocommit.

Other option is that we do Hive commit with Presto commit and only allow access of Hive transactional tables under autocommit mode.

Reply:
Sorry for the long delay, was out of town.  I've been playing with the code, and I think the last commit here will do it: https://github.com/prestosql/presto/pull/2067

Just waiting for CI to finish to be sure.

Reply:
CI failure was due to my changes, have fixed that as well and squashed your commit